pkg/tsdb/azuremonitor/util/util.go

```diff
@@ -0,0 +1,64 @@
+package util
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/tsdb/azuremonitor/types"
+)
+
+func getBackendUserFromContext(ctx context.Context) (*backend.User, bool) {
+	if val := ctx.Value(backend.User{}); val != nil {
+		user, ok := val.(*backend.User)
+		return user, ok
+	}
+	return nil, false
+}
+
+func InstrumentQueryDataRequest(ctx context.Context, req *http.Request, dsInfo types.DatasourceInfo, client *http.Client, logger log.Logger, logMessage string) (*http.Response, error) {
+	logParams := []interface{}{
+		"url", req.URL.Host,
+		"method", req.Method,
+		"path", req.URL.Path,
+		"cloud", dsInfo.Cloud,
+		"datasourceID", dsInfo.DatasourceID,
+		"subId", dsInfo.Settings.SubscriptionId,
+		"orgId", dsInfo.OrgID,
+	}
+
+	start := time.Now()
+	req.URL = nil
+	res, err := client.Do(req)
+	elapsed := time.Since(start)
+
+	if err != nil {
+		logParams = append(logParams, "status", "internal error")
+		logParams = append(logParams, "error", err.Error())
+	} else {
+		logParams = append(logParams, "status", res.StatusCode)
+	}
+
+	logParams = append(logParams, "duration", elapsed)
+
+	if dsInfo.Credentials != nil {
+		logParams = append(logParams, "authType", dsInfo.Credentials.AzureAuthType())
+	}
+
+	user, ok := getBackendUserFromContext(ctx)
+	if ok {
+		logParams = append(logParams, "uname", user.Email)
+	}
+
+	traceID := tracing.TraceIDFromContext(ctx, false)
+	if traceID != "" {
+		logParams = append(logParams, "traceID", traceID)
+	}
+
+	logger.Info(logMessage, logParams...)
```

Comment:
Nit. Not sure about info level. Can be very "spammy". Maybe debug is more suitable?

Reply:
I agree this can lead to a lot of logs that users may not want. My suggestion is to use the current tracing to achieve the same. You can see that some fields are being already tracked (for example, for Metrics): 

https://github.com/grafana/grafana/blob/main/pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go#L220

That way, only people with the tracing environment set up will get the information.